pkg/logs/schedulers/ad/scheduler.go

```diff
@@ -144,11 +145,14 @@ func (s *Scheduler) toSources(config integration.Config) ([]*sourcesPkg.LogSourc
 	var configs []*logsConfig.LogsConfig
 	var err error
 
-	switch config.Provider {
-	case names.File:
+	switch {
+	case config.Provider == names.File:
 		// config defined in a file
 		configs, err = logsConfig.ParseYAML(config.LogsConfig)
-	case names.Container, names.Kubernetes, names.KubeContainer:
+	case config.Provider == names.Container,
+		config.Provider == names.Kubernetes,
+		config.Provider == names.KubeContainer,
+		config.Provider == names.RemoteConfig && pkgconfig.Datadog.GetBool("remote_configuration.agent_integrations.allow_log_config_scheduling"):
```

Comment:
We should provide a separate error message when trying to add a log config into an agent where the feature is supported, but disabled, probably at the warning level since it is not technically an error on the agent part.

Also, can we separate this branch from the containers, as the comment below about pod annotations does not apply to remote config.

Reply:
Good points! Updated 👍